usvm-core/src/main/kotlin/org/usvm/constraints/PathConstraints.kt

```diff
@@ -23,35 +24,48 @@ import org.usvm.uctx
  */
 open class UPathConstraints<Type>(
     protected val ctx: UContext<*>,
+    protected open var ownership: MutabilityOwnership,
     protected val logicalConstraints: ULogicalConstraints = ULogicalConstraints.empty(),
     /**
      * Specially represented equalities and disequalities between objects, used in various part of constraints management.
      */
-    protected val equalityConstraints: UEqualityConstraints = UEqualityConstraints(ctx),
+    protected val equalityConstraints: UEqualityConstraints = UEqualityConstraints(ctx, ownership),
     /**
      * Constraints solved by type solver.
      */
     val typeConstraints: UTypeConstraints<Type> = UTypeConstraints(
+        ownership,
         ctx.typeSystem(),
         equalityConstraints
     ),
     /**
      * Specially represented numeric constraints (e.g. >, <, >=, ...).
      */
-    protected val numericConstraints: UNumericConstraints<UBv32Sort> = UNumericConstraints(ctx, sort = ctx.bv32Sort),
-) : UMergeable<UPathConstraints<Type>, MutableMergeGuard> {
+    protected val numericConstraints: UNumericConstraints<UBv32Sort> =
+        UNumericConstraints(ctx, sort = ctx.bv32Sort, ownership)
+): UOwnedMergeable<UPathConstraints<Type>, MutableMergeGuard> {
     init {
         // Use the information from the type constraints to check whether any static ref is assignable to any symbolic ref
         equalityConstraints.setTypesCheck(typeConstraints::canStaticRefBeEqualToSymbolic)
     }
 
+    /**
+     * Recursively changes ownership for all nested data structures that use persistent maps.
+     */
+    fun changeOwnership(ownership: MutabilityOwnership) {
+        this.ownership = ownership
+        numericConstraints.ownership = ownership
```

Comment:
It is better to introduce method for ownership modification in other constraints.

Reply:
You mean function instead of changes via setter in logical, equality, etc conatraints?

Reply:
Yes